py-scripts/lf_ftp.py

```diff
@@ -1880,6 +2108,178 @@ def copy_reports_to_home_dir(self):
+                                    statuslist.append(status[i])
+        if len(clients) != 0:
+            dataframe = {
+                " Clients": clients,
+                " MAC ": macids,
+                " Channel": channels,
+                " SSID ": ssids,
+                " Mode": modes,
+                " No of times File downloaded ": downloadtimes,
+                " Average time taken to Download file (ms)": avgtimes,
+                " Bytes-rd (Mega Bytes) ": readbytes
+            }
+            if self.expected_passfail_val or self.csv_name:
+                dataframe[" Expected value of no of times file downloaded"] = input_list
+                dataframe[' Status '] = statuslist
+
+            return dataframe
+        else:
+            return 0
+
+
+def validate_args(args):
+    """Validate CLI arguments."""
+    # Get group and profile values from arguments and convert comma-separated strings into lists
+    if args.group_name:
+        selected_groups = args.group_name.split(',')
+    else:
+        selected_groups = []  # Default to empty list if group name is not provided
+    if args.profile_name:
+        selected_profiles = args.profile_name.split(',')
+    else:
+        selected_profiles = []  # Default to empty list if profile name is not provided
+
+    if args.device_csv_name and args.expected_passfail_value:
+        logger.warning("Enter either --device_csv_name or --expected_passfail_value")
+        exit(1)
+    if args.clients_type == 'Real' and args.config and args.group_name is None:
+        if args.ssid and args.security and args.security.lower() == 'open' and (args.passwd is None or args.passwd == ''):
+            args.passwd = '[BLANK]'
+        if args.ssid is None:
+            logger.info('Specify SSID for confiuration, Password(Optional for "open" type security) , Security')
+            exit(1)
+        elif args.ssid and args.passwd:
+            if args.security is None:
+                logger.info('Security must be provided when SSID and Password specified')
+                exit(1)
+            elif args.security.lower() == 'open' and args.passwd != '[BLANK]':
+                logger.info("For a open type security there will be no password or the password should be left blank (i.e., set to '' or [BLANK]).")
+                exit(1)
+        elif args.ssid and args.passwd == '[BLANK]' and args.security and args.security.lower() != 'open':
+            logger.info('Please provide valid passwd and security configuration')
+            exit(1)
+
+    if len(selected_groups) != len(selected_profiles):
+        logger.warning("Number of groups should match number of profiles")
+        exit(1)
+    elif (args.group_name and (args.file_name is None or args.profile_name is None)) or (args.file_name and (args.group_name is None or args.profile_name is None)) or (args.profile_name and (args.group_name is None or args.file_name is None)):
```

Comment:
Please break up this conditional somehow, it's way too long. Making checks for `thing is None` can also be done by `not thing`.